mpi/docs/mpi_neighbor_allgather/example_1.c

```diff
@@ -0,0 +1,94 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <mpi.h>
+
+/**
+ * @brief Illustrates how to use MPI_Neighbor_allgather collective operator.
+ * @details An application where each MPI process creates a MPI_Graph topology and sends its neighbor a single element of type MPI_INT,
+ * with the value being the rank of the MPI process in the graph communicator.
+ * The rank is sent to every neighbour using MPI_Neighbor_allgather collective operator. 
+ **/
+
+
+/** 
```

Comment:
Feel free to remove this portion, so that it becomes a single documentation block, with a few blank lines if needed.